src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -74,32 +95,27 @@ def get_condition_result(self) -> DataConditionResult:
 
         return None
 
-    def evaluate_value(self, value: float | int) -> DataConditionResult:
+    def evaluate_value(self, value: Any) -> DataConditionResult:
         # TODO: This logic should be in a condition class that we get from `self.type`
         # TODO: This evaluation logic should probably go into the condition class, and we just produce a condition
         # class from this model
         try:
-            condition = Condition(self.condition)
+            condition_op_key = Condition(self.type)
         except ValueError:
             logger.exception(
                 "Invalid condition", extra={"condition": self.condition, "id": self.id}
             )
             return None
 
-        op = condition_ops.get(condition)
+        op = condition_ops.get(condition_op_key)
         if op is None:
             logger.error("Invalid condition", extra={"condition": self.condition, "id": self.id})
             return None
 
-        try:
-            comparison = float(self.comparison)
-        except ValueError:
-            logger.exception(
-                "Invalid comparison value", extra={"comparison": self.comparison, "id": self.id}
-            )
-            return None
+        if self.type in FILTER_VALUE_BY_CONDITIONS:
```

Comment:
🤔 it might be nicer to have a the `DataCondition` base class, then have another class that extends this; `GroupEventDataCondition` - that is always passed a `GroupEvent` as the `value` in evaluate - that would allow us to be explicit about some of the mappings setup ehre.